src/figuresClasses.ts

```diff
@@ -2,10 +1,114 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function roundTo2FloatingPoints(num: number): number {
+  return Math.floor(num * 100) / 100;
+}
+
+function validateTriangle(a: number, b: number, c: number): void {
+  if (!a || !b || !c || a <= 0 || b <= 0 || c <= 0) {
+    throw new Error('Triangle side should be greater than 0');
+  }
+
+  const sorted: number[] = [a, b, c].sort((side1, side2) => side2 - side1);
+
+  if (sorted[0] >= sorted[1] + sorted[2]) {
+    throw new Error(
+      // eslint-disable-next-line max-len
+      'Triangle longest side should not be equal or greater the sum of the remaining sides',
+    );
+  }
+}
 
-export class Circle implements Figure {}
+export class Triangle implements Figure {
+  public readonly shape: Shape;
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    validateTriangle(a, b, c);
+    // this.a = a;
+    // this.b = b;
+    // this.c = c;
+    this.shape = 'triangle';
+    // this.color = color;
```

Comment:
The line for initializing `this.color` is commented out. Since `color` is a parameter, it should be assigned to `this.color` to ensure the property is set correctly.